es7/2017-01/jan-26.md

```diff
@@ -0,0 +1,1312 @@
+
+MM: To be positive, this API makes it much less hard than it could have been
+
+DH: But the fact that you can share objects gives a lot of power. Realms plus Proxy gives you a lot of power to enforce boundaries, but you still have to be careful when setting it up.
+
+AWB: You'd probably want to instantiate global.Proxy rather than Proxy/
+
+BF: is the global going to have Proxy hooks?
+
+DH: You could set a handler to the global object, yes, that's part of the design.
+
+BF: The global object has a proxy
+
+MM: the Window vs WindowProxy thing is actually orthogonal from EcmaScript Proxies. It is an unfortunate name coincidence. A so-called WindowProxy is not a Proxy.
+
+BT: I'm curious what your thoughts are on what it would take to do on-the-fly transpilation. Would this be a combination of the Loader's translate hook plus
+
+DH: I don't think the Loader API will be necessary on the web platform at all. I don't think I'm ready to announce that I'll stop work on the loader, but I think we can get all the benefits of the Loader with a radically simpler API. But today's not the day to talk it through.
+
+MM: I am shocked; we'll take it offline.
+
+BT: But you do think the Realm APi can be used to support the on-the-fly transpilation scenario? Which will be used when you run a top-level script?
+
+MM: it has to be a different one, as the spec provides different rules for "eval code" compared to "script code".
+
+DH: I don't want to talk through all the details, but agree that hooks are an important part of it, and that we should investigate the hooks as a way to look into module loading as well.
+
+BT: This makes a lot of sense to me, that any source text is associated with just one realm--putting hooks here seems feasible
+
+DE: Would it be reasonable to make an MVP Realm and pull off some of these features into a follow-on proposal?
+
+DH: All of these features are necessary to fully virtualize an execution environment, which is the goal of this proposal. Does that seem reasonable?
+
+DE: I don't understand what's required to virtualize an execution environment exactly.
+
+AWB: It's important to think through the cross-cutting concerns first.
+
+BF: This ties into the global module registry; I am not sure it's needed.
+
+DH: I think the registry could be in userspace
+
+AWB: The Realm API was previously in the ES2015 spec, and it was pulled out due to exactly this issue, interactions between module loading and realm hooks.
+
+#### Conclusion/Resolution
+- Stage 1
+
+
+## Meeting Minutes
+
+AWB: By now everyone should have had a chance to review the minutes of the last meeting. Approve the minutes?
+
+WH: Yes, but we need to fix the deleted agenda lines.
+
+AWB: Yes, we'll fix those.
+
+#### Conclusion/Resolution
+
+Minutes are approved.
+
+
+
+## global
```

Comment:
Found numerous typos in this section, too many to individually point out. (The typing seems rushed.)